api_generator/lib/api_generator.rb

```diff
@@ -37,6 +38,20 @@ def generate(gem_folder = '../', version: nil, namespace: nil, actions: nil)
     IndexGenerator.new(gem_folder.join('lib/opensearch'), namespaces).generate
   end
 
+  def generate_low_level(gem_folder = '../')
```

Comment:
An idea: in Ruby it would be idiomatic to call methods that have side effects with a `!`, so `generate!` or `generate_low_level!`.

Reply:
The convention for bang methods I've been following is very different: A bang method is usually accompanied by a non-bang method. The bang method makes direct changes to the object it acts on while the non-bang equivalent doesn't. For example `array.sort` vs `array.sort!`.

What side effects are we introducing? Those methods generates ruby files, which is what's expected.